mount.go

```diff
@@ -143,37 +144,106 @@ func parseMountFlagsAndOptions(optionList []string) (int, string, error) {
 	return flags, strings.Join(options, ","), nil
 }
 
-func addMounts(mounts []*pb.Storage) ([]string, error) {
+func removeMounts(mounts []string) error {
+	for _, mount := range mounts {
+		if err := syscall.Unmount(mount, 0); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// storageDriversHandler is the type of callback to be defined to handle every
+// type of storage driver.
+type storageDriversHandler func(storage pb.Storage) (string, error)
+
+// storageDriversHandlerList lists the supported drivers.
+var storageDriversHandlerList = map[string]storageDriversHandler{
```

Comment:
How about making an `announce()` function that is called after `initLogger()` and that adds log entries for:

- `agentLog.WithField("version", version).Info()` (move it from `initLogger()`).
- `agentLog.WithField("storage-drivers", ...)` which shows the keys of this map.

Reply:
Yes, it's just an idea - when the agent starts it could "announce" a few details about itself in the log (version and what types of devices it can handle).

Reply:
Sure, sounds good.